src/sentry/static/sentry/app/views/settings/organizationDeveloperSettings/sentryApplicationDetails.jsx

```diff
@@ -0,0 +1,133 @@
+import JsonForm from 'app/views/settings/components/forms/jsonForm';
+import SettingsPageHeader from 'app/views/settings/components/settingsPageHeader';
+import TextCopyInput from 'app/views/settings/components/forms/textCopyInput';
+import sentryApplication from 'app/data/forms/sentryApplication';
+import getDynamicText from 'app/utils/getDynamicText';
+import ApplicationScopes from './applicationScopes';
+
+class SentryApplicationDetails extends AsyncView {
+  static contextTypes = {
+    router: PropTypes.object.isRequired,
+  };
+
+  getDefaultState() {
+    return {
+      loading: true,
+      error: false,
+      app: null,
+      errors: {},
+    };
+  }
+
+  getEndpoints() {
+    let {appSlug} = this.props.params;
+    if (appSlug) {
+      return [['app', `/sentry-apps/${appSlug}/`]];
+    }
+    return [];
+  }
+
+  getTitle() {
+    return 'Sentry Application Details';
+  }
+
+  handleScopeChange = (onChange, onBlur, scope, scopes, e) => {
+    onChange(scopes, e);
+    onBlur(scopes, e);
+  };
+
+  onSubmitSuccess = data => {
+    const {orgId} = this.props.params;
+    browserHistory.push(`/settings/${orgId}/developer-settings/`);
+  };
+
+  renderBody() {
+    const {orgId} = this.props.params;
+    const {app} = this.state;
+    let method = app ? 'PUT' : 'POST';
+    let endpoint = app ? `/sentry-apps/${app.slug}/` : '/sentry-apps/';
+    return (
+      <div>
+        <SettingsPageHeader title={this.getTitle()} />
+        <Form
+          apiMethod={method}
+          apiEndpoint={endpoint}
+          allowUndo
+          initialData={{organization: orgId, ...app}}
+          onSubmitSuccess={this.onSubmitSuccess}
+          onSubmitError={err => addErrorMessage('Unable to save change')}
+        >
+          <Box>
+            <JsonForm location={this.props.location} forms={sentryApplication} />
+            <Panel>
+              <PanelHeader>{t('API Scopes')}</PanelHeader>
+              <PanelBody>
+                <FormField
+                  name="scopes"
+                  inline={false}
+                  getData={data => ({scopes: data})}
+                  required
+                >
+                  {({onChange, onBlur}) => (
+                    <ApplicationScopes
+                      onToggle={this.handleScopeChange.bind(this, onChange, onBlur)}
+                      scopes={app && app.scopes ? app.scopes : []}
+                    />
+                  )}
+                </FormField>
+              </PanelBody>
+            </Panel>
+            {app && (
+              <Panel>
+                <PanelHeader>{t('Credentials')}</PanelHeader>
+                <PanelBody>
+                  <FormField name="clientID" label="Client ID" overflow>
+                    {({value}) => {
+                      return (
+                        <div>
+                          <TextCopyInput>
+                            {getDynamicText({value, fixed: 'PERCY_CLIENT_ID'})}
```

Comment:
What's PERCY doing here?

Reply:
copied that from https://github.com/getsentry/sentry/blob/master/src/sentry/static/sentry/app/views/settings/account/apiApplicationDetails.jsx#L67
assumed was needed for tests?

Reply:
Yeah it's for our percy snapshots, it hides dynamic content based on env vars.

Reply:
Cool. TIL.